spring-kafka/src/main/java/org/springframework/kafka/support/serializer/JsonDeserializer.java

```diff
@@ -310,6 +310,17 @@ public void setTypeMapper(Jackson2JavaTypeMapper typeMapper) {
 		}
 	}
 
+	/**
+	 * Configure the ObjectMapper before it is used to construct the ObjectWriter.
+	 * @param objectMapper configurable (or replaceable) ObjectMapper instance
+	 * @return configured (or replaced) ObjectMapper instance
+	 * @since 2.7.1
+	 */
+	public ObjectMapper customizeObjectMapper(ObjectMapper objectMapper) {
```

Comment:
I don't think this is what @garyrussell has meant in the issue.
We can't call non-`final` methods from the ctors.
Probably the point was to inject an `ObjectMapperCustomizer` into such a serialize impl...

Reply:
Good catch @artembilan ; yes a `Function<ObjectMapper, ObjectMapper>` setter would be safer.

Reply:
The alternative is to stop creating the ObjectWriter in the ctor and instead construct in the serialize() function (which is what JsonDeserializer does currently).  If this were done, then there's probably a better way to just hook into the configure() function.  The root cause was that the ObjectWriter was private, and there seemed no other way that wasn't brittle and likely to break in future versions.  If that's the preferred change, I can re-look at the implementation.

Reply:
As an alternative, I added `Map<String, ?>` to the customize function, and call it first thing from the configure() function instead of calling in the constructor.  I'm assuming that everywhere a serializer is instantiated, the configure() function will get called appropriately.  The other bit is to create the writer in serialize().  updated patch shortly.

Reply:
Thanks; we have a day off tomorrow so it will probably be next week before we continue.

Thanks again.